### PR TITLE
Bump to Dart SDK 1.22.0-dev.5.0

### PR DESCRIPTION
**Pull request by pq**
* updates to Dart SDK `1.22.0-dev.5.0`
* migrates flutter analysis code to new analyzer APIs
* works around analyzer `FutureOr` type resolution problem (https://github.com/dart-lang/sdk/issues/28285)

Note that we'll want to time landing this w/ @cbracken's updates to the `coverage` package.

cc @bwilkerson @Hixie @abarth 

FYI: @cbracken

### CODE REVIEW

**Comment by cbracken** [comment]
Did some quick digging against the 1.22.0-dev5.0 SDK release, and the new Observatory auth work isn't yet enabled by default, so don't worry about waiting on the `package:coverage` update for this.

**Comment by pq** [comment]
Ah!  Thanks @cbracken .

In that case, I'll wait for a review and land! 👍

**Comment by pq** [comment]
@abarth , @Hixie : could one of you take a quick look?

Thanks!

File: packages/flutter_tools/lib/src/dart/analysis.dart [hunk 1]
```diff
@@ -274,7 +261,12 @@ class _StdLogger extends Logger {
   @override
   void logError(String message, [Exception exception]) =>
       errorSink.writeln(message);
+
   @override
-  void logInformation(String message, [Exception exception]) =>
+  void logInformation(String message, [Exception exception]) {
+    // TODO: remove once addressed in analzyer (http://dartbug.com/28285)
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/dart/analysis.dart]: nit: TODO syntax is `// TODO(username): message`

File: packages/flutter_tools/lib/src/dart/analysis.dart [hunk 2]
```diff
@@ -274,7 +261,12 @@ class _StdLogger extends Logger {
...
+    if (message != 'No definition of type FutureOr') {
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/dart/analysis.dart]: nit: no braces

**Comment by Hixie** [comment]
Is this something that it makes sense to have tests for?

**Comment by Hixie** [comment]
RSLGTM

**Comment by pq** [comment]
FWIW: holding off on this pending investigation of analysis server issues with this integration.  (In progress!)

File: packages/flutter_tools/lib/src/dart/analysis.dart (hunk 2 above)
Comment (pq) [inline packages/flutter_tools/lib/src/dart/analysis.dart]: Done!

File: packages/flutter_tools/lib/src/dart/analysis.dart (hunk 1 above)
Comment (pq) [inline packages/flutter_tools/lib/src/dart/analysis.dart]: Done!

**Review by pq — COMMENTED** [review]
Thanks!

**Comment by pq** [comment]
> Is this something that it makes sense to have tests for?

👍  Absolutely.  Integration tests here would be great.

**Comment by yjbanov** [comment]
I'm going to roll back this change because it broke a lot of our integration tests (see https://github.com/flutter/flutter/pull/7435, and https://flutter-dashboard.appspot.com/build.html)

**Comment by pq** [comment]
Thanks @yjbanov!

**Comment by pq** [comment]
@yjbanov  (or anyone): any thoughts on why these tests were falling over?

**Comment by yjbanov** [comment]
@pq The [log file](https://flutter-dashboard.appspot.com/api/get-log?ownerKey=ahNzfmZsdXR0ZXItZGFzaGJvYXJkclgLEglDaGVja2xpc3QiOGZsdXR0ZXIvZmx1dHRlci9lZjI1YTIwMzM3ODFhOWFiMzE5NWJkZGQ4NTcyZTlhNDQ5ZDFjN2U0DAsSBFRhc2sYgICAgIDkqQgM) points at [this line in logger.dart](https://github.com/flutter/flutter/blob/016b5ab0cc9f69377cd2a1ed418687aa87e8e197/packages/flutter_tools/lib/src/base/logger.dart#L224):

```dart
  set singleCharMode(bool value) {
    stdin.lineMode = !value;  // THIS FAILS
  }
```

The error message is:

```
StdinException: Error setting terminal line mode, OS Error: Inappropriate ioctl for device, errno = 25
```

**Comment by devoncarew** [comment]
Hopefully address here: https://github.com/flutter/flutter/pull/7439

**Comment by pq** [comment]
@devoncarew : does that mean we're in a position to try this again? (And thanks!)

**Comment by devoncarew** [comment]
Yes, I think so. We may want to tweak the change to `stdin.lineMode`, but this PR hopefully won't get hung up there.

**Pull request merged**